integration/query_test.go

```diff
@@ -1280,3 +1324,114 @@ func TestQueryCommandGetMore(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryCommandGetMoreConnection(t *testing.T) {
+	t.Parallel()
+
+	// set minPoolSize and maxPoolSize 1 to ensure only one pool exists duration of the test,
+	// which forces a client to use a single connection pool
+	s := setup.SetupWithOpts(t, &setup.SetupOpts{
+		ExtraOptions: url.Values{
+			"minPoolSize": []string{"1"},
+			"maxPoolSize": []string{"1"},
+		},
+	})
+
+	collection1 := s.Collection
+	databaseName := s.Collection.Database().Name()
+	collectionName := s.Collection.Name()
+
+	docs := generateDocuments(0, 5)
+	_, err := collection1.InsertMany(s.Ctx, docs)
+	require.NoError(t, err)
+
+	t.Run("SameClient", func(t *testing.T) {
+		// Do not run subtests in t.Parallel() to reduce the occurrence
```

Comment:
That should eliminate those occurrences, not just reduce them

Reply:
In practice, I'm not 100% sure that it eliminates them. But that is the hope yes.

Reply:
The only case when it will not be fixed is if the client closes and re-opens that single connection. If it does, we could use it as we want in all those tests. With `maxIdleTimeMS=0` (which is a default, but I think we could pass that explicitly to make things a bit more clear), that should not happen.

Reply:
Thanks for looking into this. Great.